vm/src/stdlib/os.rs

```diff
@@ -191,7 +193,15 @@ pub(crate) fn fspath(
     };
     let obj = match match1(obj)? {
         Ok(pathlike) => return Ok(pathlike),
-        Err(obj) => obj,
+        Err(obj) => match PyBufferRef::try_from_borrowed_object(vm, &obj) {
```

Comment:
The reason why I try to implement this part of code inside trait `TryFromObject` in `PyPathLike` is because in CPython `os.fspath` doesn't support class `bytearray` or `memoryview`.

Maybe we could add a flag like `allow_buffer_like` in the function and reject buffer like object in `os.fspath` to match CPython behavior?

Reply:
I didn't know it very well but fspath and PathLike seem different. Thank you!